integration/cri-api/pkg/apis/services.go

```diff
@@ -62,7 +62,7 @@ type ContainerManager interface {
 	// ContainerStatus returns the status of the container.
 	ContainerStatus(containerID string) (*runtimeapi.ContainerStatus, error)
 	// UpdateContainerResources updates the cgroup resources for the container.
-	UpdateContainerResources(containerID string, resources *runtimeapi.LinuxContainerResources) error
+	UpdateContainerResources(containerID string, resources *runtimeapi.LinuxContainerResources, windowsResources *runtimeapi.WindowsContainerResources) error
```

Comment:
what about the [annotations](https://github.com/kubernetes/cri-api/blob/d059f89d4bb00d7f29a89808f43e063ce35b50de/pkg/apis/runtime/v1alpha2/api.proto#L1139)? Should we also pass those through here?

Reply:
Anything under ``integration/`` is meant for testing only (``integration`` and ``cri-integration`` tests). Basically, this will allow us to write tests that will cover Windows cases as well. In fact, you can see that ``truncindex_test.go`` is updated to test ``UpdateContainerResources`` as well.

Additionally, on the server side, the annotations are not being used / referred [1], so that won't change anything, and it is outside the scope of this PR as well.

[1] https://github.com/containerd/containerd/blob/8493cd1a506b791fef29bbae544a1074f6b7bb6d/pkg/cri/server/container_update_resources_linux.go#L39

Reply:
True, but the annotations weren't added until the windows resources field was added, so it would make sense that the annotations aren't being used yet. I'm fine with it being a separate PR though.